pkg/gofr/handler_test.go

```diff
@@ -24,17 +24,20 @@ var (
 func TestHandler_ServeHTTP(t *testing.T) {
 	testCases := []struct {
 		desc       string
+		method     string
 		data       interface{}
 		err        error
 		statusCode int
 	}{
-		{"data is nil and error is nil", nil, nil, http.StatusOK},
-		{"data is mil, error is not nil", nil, errTest, http.StatusInternalServerError},
+		{"method is get, data is nil and error is nil", http.MethodGet, nil, nil, http.StatusOK},
+		{"method is get, data is mil, error is not nil", http.MethodGet, nil, errTest, http.StatusInternalServerError},
+		{"method is post, data is nil and error is nil", http.MethodPost, nil, nil, http.StatusCreated},
+		{"method is delete, data is nil and error is nil", http.MethodDelete, nil, nil, http.StatusNoContent},
```

Comment:
Since, data is nil for every case, so either we can include tests with data not being nil or we can remove the field and assume it `nil` by default.